docs/reference/triggers/kafka.md

```diff
@@ -1,110 +1,315 @@
+- **Static mode** (`"static"`) - In this mode, partitions are allocated statically to specific workers; each worker is assigned to handle the messages for specific partitions. For example, if the replica is handling 20 partitions and has 5 workers - partitions 0&ndash;3 are handled by worker 0, partitions 3&ndash;6 by worker 1, ..., and partitions 16&ndash;19 by worker 4. The benefit and cost of this mode are inverse to the `"pool"` mode: it's entirely possible to encounter stalled processing despite having available workers (because the available workers aren't allocated to the busy partitions), but it's guaranteed that each partition is always handled by the same worker.
+
+<a id="multiple-topics"></a>
+### Multiple topics
+
+Up until now, the overview discussed partitions and workers. But a Nuclio replica can also read from multiple topics. A Nuclio replica can use its workers to handle the partitions of multiple topics instead of only those of a single topic.
+For example, if your replica has 10 workers and is configured to handle 10 topics, each with 100 partitions, the replica is essentially using 10 workers to handle 1,000 partitions.
+
+<a id="config-params"></a>
+## Configuration parameters
+<!-- See https://kafka.apache.org/documentation/#consumerconfigs + types and
+  default values in pkg/processor/trigger/kafka/types.go. -->
+
+Use the following trigger attributes for basic configurations of your Kafka trigger.
+You can configure each attribute either in the `triggers.<trigger>.attributes.<attribute>` function `spec` element (for example, `triggers.myKafkaTrigger.attributes.sessionTimeout`) or by setting the matching `nuclio.io` annotation key (for example, `nuclio.io/kafka-session-timeout`); (note that not all attributes have matching annotation keys).
+For more information on Nuclio function configuration, see the [function-configuration reference](/docs/reference/function-configuration/function-configuration-reference.md).
+
+> **Note:** For more advanced configuration parameters, see the configuration sections under [How a message travels through Nuclio to the handler](#message-course-config-params) and [Rebalancing](#rebalancing-config-params). For an example, see [Configuration example](#config-example).
+
+- <a id="topic"></a>**`topic`** - The name of the topic on which to listen.
+  <br/>
+  **Type:** `string`
+
+- <a id="brokers"></a>**`brokers`** - A list of broker IP addresses.
+  <br/>
+  **Type:** `[]string`
+
+- <a id="paritions"></a>**`paritions`** - A list of partitions for which the function receives events.
+  <br/>
+  **Type:** `[]int`
+
+- <a id="consumerGroup"></a>**`consumerGroup`** - The name of the Kafka consumer group to use.
+  <br/>
+  **Type:** `string`
+
+- <a id="initialOffset"></a>**`initialOffset`** - The location (offset) within the partition at which to begin the message processing.
```

Comment:
Should indicate that this is used ONLY for when the partition has yet to be read. For example, if I create a new consumer group and read from partition 0 - it will read from earliest/latest as configured here. But if I connect to an existing consumer group and read from partition 0 which was already read until offset X - I will not use this configuration variable at all but rather read from offset X+1

Reply:
I edited the `initOffset` doc as follows — commit 60124a9:
> The location (offset) within the partition at which to begin the message processing when first reading from a partition. Currently, you can begin the processing either with the earliest or latest ingested messages.
> Note that once a partition is read from and connected to a consumer group, subsequent reads are always done from the offset at which the previous read stopped, and the `initialOffset` configuration is ignored.